### PR TITLE
Fix search type with spaces

### LINKED ISSUE #3796: SearchByType fails to find node instances when node type contains a space
### Current Behavior

Node types which have a space in the name do not appear in results for SearchByType.

### Expected Behavior

SearchByType should find/list nodes with matching search term, regardless of space in the node name.

### Steps To Reproduce

1 - From the Library on the left drag an instance of a node to a flow, where the node name contains a space.  E.g. "http request".  Deploy the flow to instantiate the node.
2 - In the Library on the left, hover over a node whose name contains a space.  E.g. The "http request" Library node.
3 - When the popup appears with description, documentation mini-icon, search mini-icon, etc click on the search icon to auto-fill the search phrase.  Search phrase will be populated with the node name, prepended by a type filter.  E.g.  "type: http request"
4 - Note that no instances appear in the search results.

(The above method will successfully find nodes when the node name does not contain spaces.)

### Example flow

```
paste your flow here
```

### Environment

Occurs in various versions.  All running on RPi via install approach recommended on nodered.org.
Environment 1
- Node-RED version: v2.1.14 (also observed on newer v2.x.x. version environments)
- Node.js version: v14.18.2
- npm version: v6.14.15
- Platform/OS: Debian Buster
- Browser: Safari, FireFox

Environment 2
- Node-RED version: v3.0.0
- Node.js version: v14.19.3
- npm version: v6.14.17
- Platform/OS: Debian Buster
- Browser: Safari, FireFox

### PR DESCRIPTION
**Pull request by Steve-Mcl**
fixes #3796

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes
Use quotes for type search to handle node names with a space

e.g. `type:debug` and `type:"mqtt out"` 

## Checklist
- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `grunt` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: packages/node_modules/@node-red/editor-client/src/js/ui/search.js [hunk 1]
```diff
@@ -106,57 +106,71 @@ RED.search = (function() {
...
-    function search(val) {
-        var results = [];
-        var typeFilter;
-        var m = /(?:^| )type:([^ ]+)/.exec(val);
-        if (m) {
-            val = val.replace(/(?:^| )type:[^ ]+/,"");
-            typeFilter = m[1];
+    function extractType(val, flags) {
+        // extracts:  type:XYZ  &  type:"X Y Z"
+        console.log(`extractType(val, flags): val:${val}`)
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/search.js:111]:
```suggestion
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/search.js [hunk 2]
```diff
@@ -106,57 +106,71 @@ RED.search = (function() {
...
         val = extractFlag(val,"invalid",flags);
         val = extractFlag(val,"unused",flags);
         val = extractFlag(val,"config",flags);
         val = extractFlag(val,"subflow",flags);
         val = extractFlag(val,"hidden",flags);
         val = extractFlag(val,"modified",flags);
-        val = extractValue(val,"flow",flags);// flow:active or flow:<flow-id>
+        val = extractValue(val,"flow",flags);// flow:current or flow:<flow-id>
         val = extractValue(val,"uses",flags);// uses:<node-id>
+        val = extractType(val,flags);// uses:<node-id>
```
Comment (knolleary) [inline packages/node_modules/@node-red/editor-client/src/js/ui/search.js:138]:
```suggestion
        val = extractType(val,flags);// type:<node-type>
```

**Review by knolleary — CHANGES_REQUESTED** [review]

File: packages/node_modules/@node-red/editor-client/src/js/ui/palette.js [hunk 3]
```diff
@@ -175,9 +175,19 @@ RED.palette = (function() {
...
                 }
 
-                var safeType = type.replace(/'/g,"\\'");
+                const safeType = type.replace(/'/g,"\\'");
+                function wrapStr(str) {
+                    if(str.indexOf(' ') >= 0) {
+                        return '"' + str + '"'
+                    }
+                    return str
+                }
```
Comment (Steve-Mcl) [inline packages/node_modules/@node-red/editor-client/src/js/ui/palette.js:184]:
```suggestion
                const wrapStr = function (str) {
                    if(str.indexOf(' ') >= 0) {
                        return '"' + str + '"'
                    }
                    return str
                }
```

Comment (Steve-Mcl) [reply packages/node_modules/@node-red/editor-client/src/js/ui/palette.js:184]: fix linting "Function declarations should not be placed in blocks. Use a function expression or move the statement to the top of the outer function"

**Review by knolleary — APPROVED** [review]

**Pull request merged**